service/dap/server_test.go

```diff
@@ -5891,6 +5891,48 @@ func TestAttachRequest(t *testing.T) {
 	})
 }
 
+func TestAttachWaitForRequest(t *testing.T) {
+	if runtime.GOOS == "freebsd" {
+		// The value of /proc/[pid]/cmdline might be wrong on FreeBSD: zero or the same as the parent process.
+		t.Skip("test skipped on freebsd")
+	}
+	runTest(t, "loopprog", func(client *daptest.Client, fixture protest.Fixture) {
+		var cmd *exec.Cmd
+		defer func() {
+			if cmd != nil {
+				// prevent the process from becoming a zombie after the waitfor fails.
+				_ = cmd.Process.Kill()
+			}
+		}()
+		runDebugSessionWithBPs(t, client, "attach",
+			// Attach
+			func() {
+				client.AttachRequest(map[string]interface{}{
+					"mode":        "local",
+					"waitFor":     fixture.Path,
+					"stopOnEntry": false,
+				})
+				// Start the program to attach to
+				cmd = execFixture(t, fixture)
```

Comment:
This is a race condition, unfortunately. Debugserver's waitfor feature ignores already existing processes, so if the AttachRequest arrives after loopprog is executed it will skip it. That is why sometimes the tests are timing out on macos.